src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/CodeActions/Razor/ExtractToComponentCodeActionProvider.cs

```diff
@@ -75,16 +75,19 @@ public Task<ImmutableArray<RazorVSInternalCodeAction>> ProvideAsync(RazorCodeAct
 
         var actionParams = CreateInitialActionParams(context, startElementNode, @namespace);        
 
-        var path = FilePathNormalizer.Normalize(actionParams.Uri.GetAbsoluteOrUNCPath());
-        var directoryName = Path.GetDirectoryName(path).AssumeNotNull();
-        var importsBuilder = new StringBuilder();
-        var usingsInImportsFile = GetUsingsInImportsFile(directoryName, importsBuilder);
-
         ProcessSelection(startElementNode, endElementNode, actionParams);
 
         var utilityScanRoot = FindNearestCommonAncestor(startElementNode, endElementNode) ?? startElementNode;
+
+        // The new component usings are going to be a subset of the usings in the source razor file
+        var usingsInFile = context.SourceText.ToString()
```

Comment:
In general in tooling we don't want to read the sourcetext directly to get syntax. It's better to use the syntax tree. In this case it will be a `CSharpStatementLiteral` where there's a `keyword` of `using` in it

Reply:
the other thing is that the text in the using may not be the full namespace. Since usings are put inside the `class` they can be relative. So if you have a component `MyApp.MyComponents.Header` and a component `MyApp.HomePage` with `@using MyComponents` then `Header` is valid to use

Reply:
https://github.com/dotnet/razor/blob/main/src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Extensions/RazorSyntaxNodeExtensions.cs#L14

Reply:
perfect, I should have checked if there was an extension or helper